src/Credentials/CredentialProvider.php

```diff
@@ -692,6 +702,52 @@ private static function loadDefaultProfiles() {
         return $profiles;
     }
 
+    public static function getAssumeRoleCredentials(
```

Comment:
Should this function be separated out into 2 functions, one that handles source_profile and one that handles credential source? They don't seem to share much in the way of logic or parameters with each other, so it could be cleaner to have one function for each.

Reply:
Sure, but source_profile doesn't have much logic so I will just add that up above instead if that is alright